addons/hr_expense/static/src/views/list.js

```diff
@@ -44,11 +44,30 @@ export class ExpenseListController extends ListController {
         return this.is_account_invoicing && records.length && records.every(record => record.data.state === 'approve') && this.isExpenseSheet;
     }
 
+    displayPayment() {
+        const records = this.model.root.selection;
+        return this.is_account_invoicing && records.length && records.every(record => record.data.state === 'post' && record.data.payment_state === 'not_paid') && this.isExpenseSheet;
+    }
+
     async onClick (action) {
         const records = this.model.root.selection;
         const recordIds = records.map((a) => a.resId);
         const model = this.model.rootParams.resModel;
-        await this.orm.call(model, action, [recordIds]);
+        let res = await this.orm.call(model, action, [recordIds]);
+        if (res && 'type' in res && res.type.startsWith('ir.actions')) {
+            if(!('views' in res)) {
+                res = Object.assign(res, {views: [[false, 'form']]});
+            }
```

Comment:
Please fix the returned action directly in the python. As discussed, It will be better for the future to fix it directly at the source. Thanks!